### PR TITLE
Feature: Better NaN

### PR DESCRIPTION
**Pull request by Saalvage**
No longer separate macros!

### CODE REVIEW

**Comment by YarikTH** [comment]
It looks like a breaking change. It should be added in addition to the old macroses if the change is planned for minor version, or it should be postponed for the next major release.

But change itself looks cool. Better result with a wat less code.

**Comment by Saalvage** [comment]
The previous NaN macros are (thankfully) not in any stable release yet (only on the dev branch) so having a breaking change should be fine. Thank you for the review!

File: doctest/parts/doctest_fwd.h [hunk 1]
```diff
@@ -1131,6 +1130,28 @@ DOCTEST_INTERFACE String toString(const Approx& in);
 
 DOCTEST_INTERFACE const ContextOptions* getContextOptions();
 
+#if DOCTEST_MSVC
+#define DOCTEST_INTERFACE_DECL
```
Comment (onqtam) [inline doctest/parts/doctest_fwd.h:1134]: Isn't this the same as what's near line 320? Seems that this is a redundant redefinition.

Comment (Saalvage) [reply doctest/parts/doctest_fwd.h:1134]: Yep! Must've missed removing this when moving it to the top

**Pull request merged**